### ISSUE TITLE
Types: make DbCollection generic

### ISSUE THREAD

**Issue by chriskrycho** [miragejs/miragejs#380]
See original discussion [here](https://github.com/miragejs/miragejs/pull/262/files#r382230917). cc. @zoltan-nz

**Comment by bnowlin** [comment]
Tldr: +1 to issue.
Wrote the following before finding this issue.

"Title: Is typing Record<string, unknown> and DBCollection's any supported with the initial typescript support (v0.1.37)? 

  What I would like from mirage's typescript support:
  -Simple (slap-in-the-face) types for the different mirage objects (Registry<T> vs models vs row of data). Right now most mirage objects show as an 'js object' with x field instead of a dedicated name/type.
  -Typing resulting objects from mirage. E.g. Record<string, unknown> & DBCollection any
  -Typing input into the record/collection allowing partial or expanded objects. (Less sure about this one)

It is possible I'm mis-understanding mirage's typing. If so this becomes a feature request for typescript examples to do the above."
